### PR TITLE
Request and Mock Data setup

### PR DESCRIPTION
**Pull request by MathyouMB**
The following pull request is an example of how I envision data flow working in this application.

In Home.svelte, you can see we simply import the request we want to make and await the return of the data. 

We also conditionally pull from "mock data" found on the frontend. This means contributors will not need to setup our backend to work with this app. In the future we can look at using Factory pattern to generate mock data.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export interface Question {
+    id: number;
+    title: string;
+    body: string;
+    format: string;
+    releaseDate: string;
+    expirationDate: string;
+    difficulty: string;
+    categories: Category[];
```
Comment (AdiChops) [inline src/types.ts:14]: would we need the img source and perhaps an alternate description too for the graphs?

File: src/types.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+export interface Question {
+    id: number;
+    title: string;
+    body: string;
+    format: string;
+    releaseDate: string;
+    expirationDate: string;
+    difficulty: string;
```
Comment (EliasJRH) [inline src/types.ts:13]: Apart from some minor naming changes which can be taken care of easily, LGTM

File: src/types.ts (hunk 1 above)
Comment (MathyouMB) [reply src/types.ts:14]: Yes, we will be adding in more properties as they are added in the backend. Currently the backend doesn't have an image_url yet, so it isn't yet reflected in the type.

File: src/types.ts (hunk 2 above)
Comment (MathyouMB) [reply src/types.ts:13]: These are the names we currently have for the columns on the backend. Which names do you think should be changed?

**Pull request merged**